lib/livebook_cli/server.ex

```diff
@@ -153,9 +155,22 @@ defmodule LivebookCLI.Server do
     |> Livebook.Utils.browser_open()
   end
 
-  defp open_from_options(base_url, _opts, [url]) do
-    base_url
-    |> Livebook.Utils.notebook_import_url(url)
+  defp open_from_options(base_url, _opts, [file_or_dir_or_url]) do
+    cond do
+      File.dir?(file_or_dir_or_url) ->
+        file_system = FileSystem.Local.new(default_path: Path.expand(file_or_dir_or_url) <> "/")
+        :persistent_term.put(:livebook_local_filesystem, file_system)
```

Comment:
Unfortunately we cannot do this. This will permanently change the checkout directory and it won't work in the case where there is already a Livebook instance running. The proper path here is to allow something like this:

    https://localhost:3000/?file=path/to/file

And then in the LiveView you check for the `?file` parameters, see if it exists, and change the file component to point to it exclusively.

Reply:
okay thanks for the heads up.  
Just to note it, this query param handling does not exist yet right?

also, this is for the file case

What about directory? Do i need to use `{set_file, ...}` like it is currently in home component with the same pattern? (query param)

Reply:
There isn't params handling but you can access them on `mount`. You will want to change the value of this argument: https://github.com/livebook-dev/livebook/blob/main/lib/livebook_web/live/home_live.ex#L23

 I think you can call `FIleSystem.File` passing a file and the local filesystem. :)

Reply:
okay thanks, i will try to go this way.  closing thir pr.